File: app/src/main/java/ssilvalucas/testeandroidv2/screen/login/LoginConfigurator.java
package ssilvalucas.testeandroidv2.screen.login;

import java.lang.ref.WeakReference;

public enum LoginConfigurator {
    INSTANCE;

    public void configure(LoginActivity activity){

        LoginRouter router = new LoginRouter();
        router.activity = new WeakReference<>(activity);

        LoginPresenter presenter = new LoginPresenter();
        presenter.output = new WeakReference<LoginActivityInput>(activity);

        LoginInteractor interactor = new LoginInteractor();
        interactor.output = presenter;
        interactor.context = activity.getApplicationContext();

        if (activity.output == null){
            activity.output = interactor;
        }
        if (activity.router == null){
            activity.router = router;
        }
    }
}

File: Bank/app/src/main/java/com/tata/bank/login/LoginConfigurator.kt
package com.tata.bank.login

import java.lang.ref.WeakReference

enum class LoginConfigurator {
    INSTANCE;

    fun configure(activity: LoginActivity) {

        val router = LoginRouter()
        router.activity = WeakReference(activity)

        val presenter = LoginPresenter()
        presenter.output = WeakReference(activity)

        val interactor = LoginInteractor()
        interactor.output = presenter

        if (activity.output == null){
            activity.output = interactor;
        }

        if (activity.router == null){
            activity.router = router;
        }
    }
}